src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory_Methods.vb

```diff
@@ -242,45 +242,94 @@ Namespace Microsoft.CodeAnalysis.Semantics
-                        End If
+            ' Group the declarations by their VariableDeclaratorSyntaxes. The issue we're compensating for here is that the
+            ' the declarations that are BoundLocalDeclaration nodes have a ModifiedIdentifierSyntax as their syntax nodes,
+            ' not a VariableDeclaratorSyntax. We want to group BoundLocalDeclarations by their parent VariableDeclaratorSyntax
+            ' nodes, and deduplicate based on that. As an example:
+            '
+            ' Dim x, y = 1
+            '
+            ' This is an error scenario, but if we just use the BoundLocalDeclaration.Syntax.Parent directly, without deduplicating,
+            ' we'll end up with two ISingleVariableDeclarations that have the same syntax node. So, we group by VariableDeclaratorSyntax
+            ' to put x and y in the same IMutliVariableDeclaration
+            Dim groupedDeclarations = declarations.GroupBy(Function(declaration)
+                                                               If declaration.Kind = BoundKind.LocalDeclaration AndAlso
+                                                                  declaration.Syntax.IsKind(SyntaxKind.ModifiedIdentifier) Then
+                                                                   Debug.Assert(declaration.Syntax.Parent.IsKind(SyntaxKind.VariableDeclarator))
+                                                                   Return declaration.Syntax.Parent
+                                                               Else
+                                                                   Return declaration.Syntax
+                                                               End If
+                                                           End Function)
 
+            Dim builder = ArrayBuilder(Of IVariableDeclaration).GetInstance()
+            For Each declarationGroup In groupedDeclarations
+                Dim first = declarationGroup.First()
+                Dim singleDeclarations As ImmutableArray(Of ISingleVariableDeclaration) = Nothing
+                Dim initializer As IVariableInitializer = Nothing
+                If first.Kind = BoundKind.LocalDeclaration Then
+                    singleDeclarations = declarationGroup.Cast(Of BoundLocalDeclaration).SelectAsArray(AddressOf GetSingleVariableDeclaration)
+
+                    ' The initializer we use for this group is the initializer attached to the last declaration in this declarator, as that's
+                    ' where it will be parsed in an error case.
+                    ' Initializer is only created if it's not the array initializer for the variable. That initializer is the initializer
+                    ' of the SingleVariableDeclaration child.
+                    Dim last = DirectCast(declarationGroup.Last(), BoundLocalDeclaration)
+                    If last.InitializerOpt IsNot Nothing AndAlso last.InitializerOpt IsNot last.ArrayCreationOpt Then
```

Comment:
>last.InitializerOpt IsNot last.ArrayCreationOpt [](start = 65, length = 47)

We should never create a node, for which this condition is true.

Reply:
Implemented the solution we talked about offline.

---
In reply to: [146972586](https://github.com/dotnet/roslyn/pull/22842#discussion_r146972586) [](ancestors = 146972586)